master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/StateMachine.java

```diff
@@ -216,16 +222,17 @@ private synchronized void loadSnapshot(SingleFileSnapshotInfo snapshot) throws I
     try {
       setLastAppliedTermIndex(snapshot.getTermIndex());
       install(snapshotFile);
-    } catch (Exception e) {
-      throw new IOException(String.format("Failed to load snapshot %s", snapshot), e);
+    } catch (IOException e) {
+      throw new IOException(String.format("Failed to load snapshot %s", snapshot), e.getCause());
```

Comment:
Emm, will `e.getCause()` return null?

How about

```suggestion
    } catch (IOException rethrow) {
      LOG.error("Failed to load snapshot {}", snapshot);
      throw rethrow;
    }
```

Reply:
Done